homeassistant/components/habitica/sensor.py

```diff
@@ -270,7 +302,94 @@ async def async_setup_entry(
         HabitipyTaskSensor(coordinator, description)
         for description in TASK_SENSOR_DESCRIPTION
     )
-    async_add_entities(entities, True)
+    eggs_added: set[str] = set()
+    hatch_potion_added: set[str] = set()
+    food_added: set[str] = set()
+
+    @callback
+    def add_entities() -> None:
+        """Add sensors for inventory items."""
+
+        nonlocal eggs_added, hatch_potion_added, food_added
+
+        inventory: list[HabitipySensorEntityDescription] = []
+
+        new_eggs = set(coordinator.data.user["items"]["eggs"].keys()) - eggs_added
+        new_potions = (
+            set(coordinator.data.user["items"]["hatchingPotions"].keys())
+            - hatch_potion_added
+        )
+        new_food = set(coordinator.data.user["items"]["food"]) - food_added
+
+        inventory.extend(
+            HabitipySensorEntityDescription(
+                key=f"Pet_Egg_{key}",
+                translation_key=HabitipySensorEntity.EGG,
+                translation_placeholders={
+                    "egg_type": coordinator.content["eggs"][key]["text"]
+                },
+                value_fn=(
+                    partial(
+                        lambda user, _, key: user["items"]["eggs"][key],
+                        key=key,
+                    )
+                ),
+                entity_picture=f"Pet_Egg_{key}.png",
+                entity_registry_enabled_default=False,
+                native_unit_of_measurement="eggs",
+                attributes_fn=partial(lambda x, _, key: {"key": key}, key=key),
+            )
+            for key in new_eggs
```

Comment:
This is creating sensors dynamically based on the API response. We don't want that, as it couples us hard with the API.

If they add or remove sensors, it will directly impact us. 

This needs to be loosely coupled.

Reply:
What do you mean by loosely coupled? Should I add code that cleans up sensors for items that are no longer in the API response? I haven't done that, because that doesn't actually happen. Once a user finds or receives a new item, the API will always return that item in the response. 

This is for example from the API response for my own account. I received the seahorse, parrot and gryphon eggs like 10 years ago and they still are on the list, although i don't have any. On the other hand, when a new item is received, the list is extended and a new sensor is created for that item. When I decided to add sensors dynamically what I had in mind was that for fresh accounts, the inventory lists will be empty at the beginning and grow with the time.

```json
      "eggs": {
        "BearCub": 15,
        "Cactus": 20,
        "Dragon": 8,
        "FlyingPig": 20,
        "Fox": 20,
        "Gryphon": 0,
        "LionCub": 20,
        "PandaCub": 20,
        "Seahorse": 0,
        "TigerCub": 20,
        "Wolf": 20,
        "Parrot": 0
      },
```